Broken module parameters parsing

**Describe the bug**
When you want to pass module parameter that is not in the format `key=value`, but is a flag, such as `--no-cloudwatch-metrics` in this [module](https://github.com/gruntwork-io/terraform-aws-security/tree/main/modules/fail2ban), you will see an error like this:
```
2024-04-11 11:38:01 [INFO] [gruntwork-install] Executing /tmp/gruntwork-script-modules/fail2ban/install.sh --no-cloudwatch-metrics no-cloudwatch-metrics
2024-04-11 11:38:01 [ERROR] [configure-fail2ban] Unrecognized argument: no-cloudwatch-metrics
```
The script is not prepared to accept a parameter other than one that has the format `key=value`. Look [here](https://github.com/gruntwork-io/gruntwork-installer/blob/afedd68470edf544ca4cc297648076418ded3663/gruntwork-install#L313-L314).

**To Reproduce**
```
gruntwork-install --module-name 'fail2ban' --module-param 'no-cloudwatch-metrics' --repo https://github.com/gruntwork-io/module-security --tag 'v0.65.5'
```

**Expected behavior**
It should work fine with all existing parameters.

--- a/gruntwork-install
+++ b/gruntwork-install
@@ -310,9 +310,13 @@ function install_script_module {
         shift
         ;;
       --module-param)
-        local module_param_key="${2%%=*}"
-        local module_param_val="${2#*=}"
-        module_params+=( "--$module_param_key" "$module_param_val" )
+        if [[ $2 == *"="* ]]; then
+          local module_param_key="${2%%=*}"
+          local module_param_val="${2#*=}"
+          module_params+=( "--$module_param_key" "$module_param_val" )
+        else
+          module_params+=( "--$2" )
+        fi
         shift
         ;;
       --download-dir)

--- a/modules/mixed-args-module/README.md
+++ b/modules/mixed-args-module/README.md
@@ -1,0 +1,3 @@
+# Mixed args module
+
+This module is used solely for running automated tests against the Gruntwork Installer. 

--- a/modules/mixed-args-module/install.sh
+++ b/modules/mixed-args-module/install.sh
@@ -1,0 +1,44 @@
+#!/bin/bash
+# A dirt simple install script that expects one madatory argument, --message-to-echo, that it echoes. The echo message can be overridden with 
+# a flag, --echo-override. This is used to test that the --module-param args in gruntwork-install can handle flags without values.
+
+set -e
+
+function assert_not_empty {
+  local -r arg_name="$1"
+  local -r arg_value="$2"
+
+  if [[ -z "$arg_value" ]]; then
+    echo "ERROR: The value for '$arg_name' cannot be empty"
+    exit 1
+  fi
+}
+
+function install {
+  local echomessage
+
+  while [[ $# > 0 ]]; do
+    local key="$1"
+
+    case "$key" in
+      --message-to-echo)
+        echomessage="$2"
+        shift
+        ;;
+      --echo-override)
+        echomessage="Override"
+        ;;
+      *)
+        echo "Unrecognized argument: $key"
+        exit 1
+        ;;
+    esac
+
+    shift
+  done
+
+  assert_not_empty "--message-to-echo" "$echomessage"
+  echo "$echomessage"
+}
+
+install "$@"

--- a/test/integration-test.sh
+++ b/test/integration-test.sh
@@ -22,6 +22,9 @@ configure-ecs-instance --help
 echo "Using gruntwork-install to install a module from the gruntwork-install repo and passing args to it via --module-param"
 gruntwork-install --module-name "dummy-module" --repo "https://github.com/gruntwork-io/gruntwork-installer" --tag "v0.0.25" --module-param "file-to-cat=$SCRIPT_DIR/integration-test.sh"
 
+echo "Using gruntwork-install to install a module from the gruntwork-install repo and passing mixed args to it via --module-param"
+gruntwork-install --module-name "mixed-args-module" --repo "https://github.com/gruntwork-io/gruntwork-installer" --ref "fix/mixed-args" --module-param "message-to-echo=Hello" --module-param "echo-override"
+
 echo "Using gruntwork-install to install a module from the gruntwork-install repo with branch as ref"
 gruntwork-install --module-name "dummy-module" --repo "https://github.com/gruntwork-io/gruntwork-installer" --ref "for-testing-dont-delete" --module-param "file-to-cat=$SCRIPT_DIR/integration-test.sh"
 

